llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapIoImpl.java

```diff
@@ -91,143 +74,148 @@
-      }
+      debugDumpComponents.add(metadataCache);
       debugDumpComponents.add(allocator);
     } else {
       this.allocator = new SimpleAllocator(conf);
-      fileMetadataCache = null;
-      SimpleBufferManager sbm = new SimpleBufferManager(allocator, cacheMetrics);
-      bufferManager = bufferManagerOrc = bufferManagerGeneric = sbm;
-      dataCache = sbm;
+      final SimpleBufferManager sbm = new SimpleBufferManager(allocator, cacheMetrics);
+      this.bufferManager = bufferManagerOrc = bufferManagerGeneric = sbm;
+      this.dataCache = sbm;
+      this.debugDumpComponents.add(sb -> sb.append("LLAP IO allocator is not in use!"));
+      this.fileMetadataCache = null;
       this.memoryManager = null;
-      debugDumpComponents.add(new LlapIoDebugDump() {
-        @Override
-        public void debugDumpShort(StringBuilder sb) {
-          sb.append("LLAP IO allocator is not in use!");
-        }
-      });
+      serdeCache = null;
+      metadataCache = null;
     }
-    // IO thread pool. Listening is used for unhandled errors for now (TODO: remove?)
-    int numThreads = HiveConf.getIntVar(conf, HiveConf.ConfVars.LLAP_IO_THREADPOOL_SIZE);
-    executor = new StatsRecordingThreadPool(numThreads, numThreads, 0L, TimeUnit.MILLISECONDS,
-        new LinkedBlockingQueue<Runnable>(),
-        new ThreadFactoryBuilder().setNameFormat("IO-Elevator-Thread-%d").setDaemon(true).build());
-    FixedSizedObjectPool<IoTrace> tracePool = IoTrace.createTracePool(conf);
-    // TODO: this should depends on input format and be in a map, or something.
-    this.orcCvp = new OrcColumnVectorProducer(
-        metadataCache, dataCache, bufferManagerOrc, conf, cacheMetrics, ioMetrics, tracePool);
-    this.genericCvp = isEncodeEnabled ? new GenericColumnVectorProducer(
-        serdeCache, bufferManagerGeneric, conf, cacheMetrics, ioMetrics, tracePool) : null;
-    LOG.info("LLAP IO initialized");
 
-    registerMXBeans();
+    // TODO: this should depends on input format and be in a map, or something.
+    this.orcCvp =
+        new OrcColumnVectorProducer(metadataCache,
+            dataCache,
+            bufferManagerOrc,
+            conf,
+            cacheMetrics,
+            ioMetrics,
+            tracePool);
+    this.genericCvp =
+        isEncodeEnabled ?
+            new GenericColumnVectorProducer(serdeCache,
+                bufferManagerGeneric,
+                conf,
+                cacheMetrics,
+                ioMetrics,
+                tracePool) :
+            null;
+    this.buddyAllocatorMXBean = MBeans.register("LlapDaemon", "BuddyAllocatorInfo", allocator);
+    LOG.info("LLAP IO successful initialized.");
   }
 
-  private void registerMXBeans() {
-    buddyAllocatorMXBean = MBeans.register("LlapDaemon", "BuddyAllocatorInfo", allocator);
+  private static int[] stringsToIntegers(String[] strings) {
+    if (strings == null) {
+      return new int[0];
+    }
+    return Arrays.stream(strings).map(x -> {
```

Comment:
not that I don't appreciate a nice stream function but I suspect this is the least efficient method to covnert integer strings to int. each instance is wrapped into an optional just fo filter out the non-parsable directly again?

Reply:
i removed the wrapping